blockchain/chain_sync/src/sync.rs

```diff
@@ -620,25 +630,28 @@ where
         let db = Arc::clone(&self.chain_store.db);
         let parent_clone = parent_tipset.clone();
         // check messages to ensure valid state transitions
-        let x = task::spawn_blocking(move || Self::check_block_msgs(db, b, &parent_clone));
+        let sm = self.state_manager.clone();
+        let x = task::spawn(Self::check_block_msgs(sm, db, b, parent_clone));
         validations.push(x);
 
-        // TODO use computed state_root instead of parent_tipset.parent_state()
-
         // block signature check
+        let (state_root, _) = self
+            .state_manager
+            .tipset_state(&parent_tipset)
+            .await
+            .map_err(|_| Error::Validation("Could not update state".to_owned()))?;
         let work_addr_result = self
             .state_manager
-            .get_miner_work_addr(&parent_tipset.parent_state(), header.miner_address());
+            .get_miner_work_addr(&state_root, header.miner_address());
+
+        // temp header needs to live long enough in static context returned by task::spawn
+        let temp_header = block.header().clone();
```

Comment:
This clone is expensive, probably worth considering not using that specific function to verify the signature (all you need is the cid bytes, so cloning the whole block header is unnecessary). This wasn't introduced in this PR so feel free to leave, just commenting because I just noticed it